src/ngraph/op/fused/batch_to_space.cpp

```diff
@@ -0,0 +1,193 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <cmath>
+#include <cstddef>
+#include <memory>
+#include <ops.hpp>
+
+#include "ngraph/builder/make_constant.hpp"
+#include "ngraph/builder/reshape.hpp"
+#include "ngraph/node.hpp"
+#include "ngraph/op/fused/batch_to_space.hpp"
+#include "ngraph/shape.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+constexpr NodeTypeInfo op::v1::BatchToSpace::type_info;
+
+ngraph::op::v1::BatchToSpace::BatchToSpace(const ngraph::Output<ngraph::Node>& data,
+                                           const ngraph::Output<ngraph::Node>& block_shape,
+                                           const ngraph::Output<ngraph::Node>& crops_begin,
+                                           const ngraph::Output<ngraph::Node>& crops_end)
+    : FusedOp({data, block_shape, crops_begin, crops_end})
+{
+    constructor_validate_and_infer_types();
+}
+
+NodeVector op::v1::BatchToSpace::decompose_op() const
+{
+    auto data = input_value(0);
+    auto block = input_value(1);
+    auto crops_begin = input_value(2);
+    auto crops_end = input_value(3);
+
+    const auto& data_shape = data.get_shape();
+
+    NODE_VALIDATION_CHECK(this,
+                          (data_shape.size() >= 2),
+                          "The data tensor with rank lower than 2 is not supported (data rank: ",
+                          data_shape.size(),
+                          ")");
+
+    const auto block_const = as_type_ptr<op::Constant>(block.get_node_shared_ptr());
+    const auto crops_begin_const = as_type_ptr<op::Constant>(crops_begin.get_node_shared_ptr());
+    const auto crops_end_const = as_type_ptr<op::Constant>(crops_end.get_node_shared_ptr());
+
+    vector<int64_t> block_values, crops_begin_values, crops_end_values;
+    block_values = block_const->cast_vector<int64_t>();
+    crops_begin_values = crops_begin_const->cast_vector<int64_t>();
+    crops_end_values = crops_end_const->cast_vector<int64_t>();
+
+    // First we have to disperse the data from batch, then rearrange them
+    // so as appropriate chunks of data where close to their destination place.
+    // Finally squeeze data from respective dimensions.
+    Shape dispersed_shape;
+    int64_t b_dim_divider = 1;
+    for (auto& el : block_values)
+    {
+        NODE_VALIDATION_CHECK(this, el > 0, "block_shape values must be greater than 0");
+        b_dim_divider *= el;
+    }
+
+    NODE_VALIDATION_CHECK(this,
+                          data_shape.at(0) % b_dim_divider == 0,
+                          "BatchToSpace: The input data's 'batch' axis size: ",
+                          data_shape.at(0),
+                          " must be completely divided by ",
```

Comment:
"must be a multiple of"  sounds a bit better.

Reply:
fixed